devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeTestsMojo.java

```diff
@@ -1,16 +1,100 @@
 package io.quarkus.maven;
 
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.ResolutionScope;
 
+import io.quarkus.bootstrap.app.CuratedApplication;
+import io.quarkus.builder.Json;
+import io.quarkus.maven.dependency.ResolvedDependency;
+import io.quarkus.runtime.LaunchMode;
+
 @Mojo(name = "generate-code-tests", defaultPhase = LifecyclePhase.GENERATE_TEST_SOURCES, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME, threadSafe = true)
 public class GenerateCodeTestsMojo extends GenerateCodeMojo {
     @Override
     protected void doExecute() throws MojoExecutionException, MojoFailureException {
         generateCode(getParentDirs(mavenProject().getTestCompileSourceRoots()),
                 path -> mavenProject().addTestCompileSourceRoot(path.toString()), true);
+
+        final String testProfile = System.getProperty("quarkus.test.integration-test-profile");
```

Comment:
Not a blocker but this can be approved by consulting other config sources.
Maven project properties would be a trivial one to add but others could be made to work as well.

Reply:
Fair point, this is a leftover from my first experiments here. I'll see if I can hook another source instead.